--- ./ksh93/src/cmd/ksh93/sh/init.c	2021-01-16 15:27:03.589079734 -0600
+++ init.c	2021-01-16 15:53:50.700864511 -0600
@@ -48,6 +48,28 @@
 #include	"lexstates.h"
 #include	"version.h"

+#ifdef BUILD_DTKSH
+#include <Dt/DtNlUtils.h>
+#include <Dt/EnvControlP.h>
+#include <X11/X.h>
+#include <X11/Intrinsic.h>
+#include <X11/IntrinsicP.h>
+#include <X11/CoreP.h>
+#include <X11/StringDefs.h>
+#include <Xm/XmStrDefs.h>
+#include <Xm/Xm.h>
+#include <Xm/Protocols.h>
+
+#include "dtksh.h"
+#include "xmksh.h"
+#include "dtkcmds.h"
+#include "xmcvt.h"
+#include "widget.h"
+#include "extra.h"
+#include "xmwidgets.h"
+#include "msgs.h"
+#endif
+
 #if _hdr_wctype
 #include	<ast_wchar.h>
 #include	<wctype.h>
@@ -1481,6 +1503,16 @@
	shp->exittrap = 0;
	shp->errtrap = 0;
	shp->end_fn = 0;
+#ifdef BUILD_DTKSH
+	int * lockedFds = LockKshFileDescriptors();
+	(void) XtSetLanguageProc((XtAppContext)NULL, (XtLanguageProc)NULL,
+                                (XtPointer)NULL);
+	DtNlInitialize();
+	_DtEnvControl(DT_ENV_SET);
+	UnlockKshFileDescriptors(lockedFds);
+
+	dtksh_init();
+#endif
	return(shp);
 }
